velocidi-sdk/src/main/kotlin/com/velocidi/Velocidi.kt

```diff
@@ -1,5 +1,81 @@
 package com.velocidi
 
-fun main(args : Array<String>) {
-    println("Welcome to Velocidi SDK")
+import android.Manifest
+import android.content.Context
+import android.util.Log
+import org.json.JSONObject
+
+
+internal open class Velocidi constructor(val config: Config, context: Context): FetchAdvertisingId<Request>(context) {
+
+    val client = HttpClient()
+
+    val appInfo = Util.getApplicationInfo(context)
+
+    init {
+        client.headers["User-Agent"] = "${appInfo.appName}/${appInfo.appVersion} ${appInfo.androidSDK} ${appInfo.device}"
+        client.defaultParams["cookies"] = "false"
+    }
+
+    override fun onFetchCompleted() {
+        client.defaultParams["aaid"] = adInfo.id
+    }
+
+     override fun handleTask(task: Request) {
+        if(!adInfo.shouldTrack) return
+
+         return when(task) {
+            is com.velocidi.Request.TrackRequest ->
+                if(config.track.enabled)
+                    client.sendRequest(HttpClient.Verb.POST, config.track.host.toString(), task.attributes)
+                else return
+            is com.velocidi.Request.MatchRequest ->
+                if(config.match.enabled)
+                    client.sendRequest(HttpClient.Verb.GET, "${config.match.host}?${task.toQueryParams()}")
+                else return
+
+        }
+    }
+
+     companion object {
+         internal lateinit var instance: Velocidi
+
+         fun start(config: Config, context: Context) {
+             if(!Util.checkPermission(context, Manifest.permission.INTERNET))
+                 Log.e(Constants.LOG_TAG, "Velocidi SDK requires Internet permission")
+
+             instance = Velocidi(config, context)
+
+         }
+
+         fun getInstance(): Velocidi? =
```

Comment:
I would prefer if we could avoid returning a Nullable here. From what I understand, the only thing that keeps us from doing that right now, is the Ad Id thing... as discussed offline previously, I would prefer to have a queue of requests, and encode the behavior of only triggering requests once the concurrent task of retrieving the ID completes.